src/Raven.Server/Documents/DatabasesLandlord.cs

```diff
@@ -1126,6 +1127,16 @@ public RavenConfiguration CreateDatabaseConfiguration(StringSegment databaseName
             return CreateDatabaseConfiguration(_serverStore, databaseRecord.DatabaseName, databaseRecord.Settings);
         }
 
+        private static bool TryGetDeletionInProgress(Dictionary<string, DeletionInProgressStatus> deletionInProgress, string databaseName, string nodeTag, out DeletionInProgressStatus status)
+        {
+            if (ShardHelper.TryGetShardNumberAndDatabaseName(ref databaseName, out int shardNumber))
```

Comment:
this `ref` scares me :)
let's change it to 
`TryGetShardNumberAndDatabaseName(string database, out string shardedDatabaseName, out int shardNumber)`

Reply:
Changed